### PR TITLE
#153971885 Create authenication flask blueprints

### PR DESCRIPTION
**Pull request by betsybeth**

### CODE REVIEW

File: app/auth/views.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+        if name and email and password:
+            if User.validate_email(email):
+
+                if name.strip() == "":
+                    response = {'message': "invalid name"}
+                    return make_response(jsonify(response)), 400
+
+                if len(password) < 8:
+                    response = {
+                        'message': 'email too short'
```
Comment (Rodgers-M) [inline app/auth/views.py:27]: return appropriate message, it should be `password too short` not `email`

**Review by Rodgers-M — COMMENTED** [review]
Good implementation of the blueprints.

You should add the `.env` file to `.gitignore`. it holds application secrets which should not be exposed to public. check [autoenv](https://pypi.python.org/pypi/autoenv/0.2.0) which will execute the `.env` and export your variables.

File: app/auth/views.py [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+    def post(self):
+        """Logout a registered user."""
+        auth_token = request.headers.get('Authorization')
+        if auth_token:
+            resp = User.decoding_token(auth_token)
+            return make_response(
+                jsonify({'message': 'successfully logout'})), 200
+        return make_response(
+            jsonify({'messsage': 'wrong token,Please check you token if correct'})), 403
```
Comment (meshnesh) [inline app/auth/views.py:88]: this would be not a good message to a user

File: app/auth/views.py [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+            response = {
+                "message": 'Password reset successful',
+                'new password': new_password
+            }
+            return make_response(jsonify(response)), 201
+        response = {'message': 'Invalid email'}
+        return make_response(jsonify(response)), 400
+
+
+class ChangePassword(MethodView):
```
Comment (meshnesh) [inline app/auth/views.py:112]: what's the use of this class if you have the rest password, if they perform the same tasks, you can combine them

File: app/models.py [hunk 4]
```diff
@@ -1,52 +1,62 @@
...
         return check_password_hash(self.password, password)
 
-    def encoding_token(self, user_id):
+    def generate_token(self, user_id):
         """Generate a token for authenication return String."""
         try:
             payload = {
-                "exp": datetime.utnow() + datetime.timedelta(seconds=10),
-                "iat": datetime.datetime.utnow(),
+                "exp": datetime.utcnow() + timedelta(seconds=10),
```
Comment (meshnesh) [inline app/models.py:42]: the expiration for a token is too little

File: run.py [hunk 5]
```diff
@@ -5,4 +5,4 @@
 app = create_app(config_name)
 
 if __name__ == '__main__':
-    app.run()
+    app.run(debug=True)
```
Comment (meshnesh) [inline run.py:8]: It is not good to set the debug here, you could use the config file for this

**Comment by meshnesh** [comment]
Good implementation of the tables and generating tokens and decoding them

File: run.py (hunk 5 above)
Comment (betsybeth) [reply run.py:8]: Okay.

File: app/auth/views.py (hunk 3 above)
Comment (betsybeth) [reply app/auth/views.py:112]: The ResetPassword class and ChangePassword class are not the same.
ChangePassword is when the user can remember the password but want to change the password
ResetPassword is when a user has completely forgotten the password

**Pull request merged**